action.yml

```diff
@@ -56,15 +56,68 @@ inputs:
     description: >-
       Reduce issue severity by one level, for non-security issues, for compatibility with GitHub's code scanning feature.
       This option will only have an effect when used in conjunction with '--format sarif'.
+  run-docker-tools:
+    required: false
+    default: "true"
+    description: >-
+      Run dockerized tools. Possible values (true|false)
+      Check the full list in https://docs.codacy.com/repositories-configure/codacy-configuration-file/#which-tools-can-be-configured-and-which-name-should-i-use
+  run-gosec:
+    required: false
+    description: >-
+      Run GoSec. Possible values (true|false)
 runs:
   using: "composite"
   steps:
-    - name: "Set Repository Variables"
+    - name: "Set Global Variables"
       shell: bash
       run: |
+        echo "CODACY_BASE_URL_OR_DEFAULT=$(if [ ${{ inputs.codacy-api-base-url }} ]; then echo "${{ inputs.codacy-api-base-url }}"; else echo "https://api.codacy.com"; fi)" >> $GITHUB_ENV
         echo "OWNER_NAME=$(echo $GITHUB_REPOSITORY | cut -d '/' -f 1)" >> $GITHUB_ENV
         echo "REPOSITORY_NAME=$(echo $GITHUB_REPOSITORY | cut -d '/' -f 2)" >> $GITHUB_ENV
         echo "ORGANIZATION_PROVIDER=$(if [ "$GITHUB_SERVER_URL" == "https://github.com" ]; then echo "gh"; else echo "ghe"; fi)" >> $GITHUB_ENV
+
+    - name: "Prepare curl authentication header"
+      shell: bash
+      run: |
+        if [ -n "${{ inputs.api-token }}" ]; then
+          echo "CURL_CODACY_AUTH_AUTHENTICATION=api-token: ${{ inputs.api-token }}" >> $GITHUB_ENV
+        elif [ -n "${{ inputs.project-token }}" ]; then
+          echo "CURL_CODACY_AUTH_AUTHENTICATION=project-token: ${{ inputs.project-token }}" >> $GITHUB_ENV
+        else
+          echo "At least one authentication method is required."
```

Comment:
Wait, if you just want to run this locally, you can't.

Reply:
What do you mean running the action locally?

Reply:
I meant without uploading results.

Reply:
ok, i see. I changed it to only apply to upload